contracts/modules/cover/MinimalBeaconProxy.sol

```diff
@@ -0,0 +1,42 @@
+import "@openzeppelin/contracts-v4/proxy/Proxy.sol";
+import "../../interfaces/IStakingPoolBeacon.sol";
+
+/**
+ * @dev This contract implements a proxy that gets the implementation address for each call from a {UpgradeableBeacon}.
+ *
+ * The beacon address is stored as an immutable field.
+ *
+ * _Available since v3.4._
+ */
+contract MinimalBeaconProxy is Proxy {
+
+  /**
+   * @dev The beacon address.
+   */
+  address immutable public beacon;
+
+  /**
+   * @dev Initializes the proxy with `beacon`.
+   *
+   * Requirements:
+   *
+   * - `beacon` must be a contract with the interface {IBeacon}.
+   */
+  constructor(address _beacon) payable {
```

Comment:
Probably makes sense to remove the `payable` keyword here as we don't plan to ever send ETH to this contract.